pol-core/pol/module/osmod.cpp

```diff
@@ -1400,6 +1460,215 @@ BObjectImp* OSExecutorModule::mf_GetEnvironmentVariable()
+        {
+          return recipient_bcc_error;
+        }
+
+        curl_easy_setopt_return_error( curl, CURLOPT_MAIL_RCPT, recipients_slist->get() );
+
+        auto get_date_header = []
+        {
+          char buffer[100];
+          std::time_t t = std::time( nullptr );
+          std::tm* tm_utc = std::gmtime( &t );
+          std::strftime( buffer, sizeof( buffer ), "Date: %a, %d %b %Y %H:%M:%S +0000", tm_utc );
+          return std::string( buffer );
+        };
+
+        std::vector<std::string> email_headers{ get_date_header(), "To: " + to_header_value,
+                                                "From: " + from->value(),
+                                                "Subject: " + subject->value() };
+
+        for ( const auto& header : email_headers )
+          *headers_slist = curl_slist_append( *headers_slist, header.c_str() );
+
+        curl_easy_setopt_return_error( curl, CURLOPT_HTTPHEADER, headers_slist->get() );
+
+        *mime = curl_mime_init( curl );
+
+        curl_mimepart* part = curl_mime_addpart( *mime );
+
+        // `curl_mime_data` copies the string into its internal structure, so using this local is
+        // okay. Ref: https://github.com/curl/curl/blob/curl-8_2_1/lib/mime.c#L1380-L1390
+        curl_mime_data( part, body->data(), CURL_ZERO_TERMINATED );
+
+        // `curl_mime_type` copies the string into its internal structure, so using this local is
+        // okay. Ref: https://github.com/curl/curl/blob/curl-8_2_1/lib/mime.c#L1461
+        curl_mime_type( part, contentType->data() );
+
+        curl_easy_setopt_return_error( curl, CURLOPT_MIMEPOST, mime->get() );
+
+        Core::networkManager.auxthreadpool->push(
+            [uoexec_w, curl_sp, recipients_slist, headers_slist, mime]()
+            {
+              CURL* curl = curl_sp.get();
+
+              auto res = curl_easy_perform( curl );
+
+              {
+                Core::PolLock lck;
+
+                if ( !uoexec_w.exists() )
+                {
+                  DEBUGLOGLN( "SendEmail Script has been destroyed" );
+                  return;
+                }
+                if ( res != CURLE_OK )
+                  uoexec_w.get_weakptr()->ValueStack.back().set(
+                      new BObject( new BError( curl_easy_strerror( res ) ) ) );
+                else
+                  uoexec_w.get_weakptr()->ValueStack.back().set( new BObject( new BLong( 1 ) ) );
+
+                uoexec_w.get_weakptr()->revive();
+              }
+            }
+
+            /* always cleanup */
+            // curl_easy_cleanup(), curl_slist_free_all(), curl_mime_free() are
+            // performed when the shared pointer deallocates
+        );
+      }
+      else
+      {
+        return new BError( "curl_easy_init() failed" );
+      }
+    }
+    else
+    {
+      return new BError( "Invalid parameter type" );
+    }
+  }
+
+  return new BError( "Invalid parameter type" );
```

Comment:
Isn't this the temp BObjectImp which gets replaced by the background thread? Why not simply return BLong(0) returning a error looks like a actual problem.

Reply:
Addressed in 0574bee2ee8a56c83cb66797038da63cc026255c